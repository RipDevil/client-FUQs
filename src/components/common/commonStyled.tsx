import styled from 'styled-components';

export const StyledSpinner = {
  SpinnerDiv: styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    z-index: 1000;
    margin: -50px 0px 0px -50px;
    max-width: 140px;
  `,
  BackgroundDiv: styled.div`
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 999;
    background-color: rgb(255, 255, 255);
  `,
  TextDiv: styled.div`
    text-align: center;
    margin-top: 1rem;
  `,
};
