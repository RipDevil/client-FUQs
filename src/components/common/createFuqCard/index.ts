export { CreateFuqCard } from './create-fuq-card';
