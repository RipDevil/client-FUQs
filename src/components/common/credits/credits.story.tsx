import React from 'react';
import { Story } from '@storybook/react';
import styled from 'styled-components';

import '@fortawesome/fontawesome-free/css/all.css';
import 'antd/dist/antd.css';
import 'index.css';

import { Credits as ToggleCredits, CreditsTypes } from 'components/common/credits/credits';
import { argTypes, args } from './lib';

// eslint-disable-next-line import/no-anonymous-default-export
export default {
  title: 'Components/Common/Credits',
  component: ToggleCredits,
  parameters: {
    controls: { sort: 'requiredFirst' },
  },
  args,
  argTypes,
};

const MockSpan = styled.span`
  div {
    position: relative !important;
  }
`;

export const Credits: Story<CreditsTypes> = ({ ...props }) => {
  return (
    <MockSpan>
      <ToggleCredits {...props} />
    </MockSpan>
  );
};
