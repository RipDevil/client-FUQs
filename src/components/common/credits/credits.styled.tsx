import styled from 'styled-components';

export const CreditsComponent = styled.div`
  position: fixed;
  bottom: 10px;
  right: 10px;
  z-index: 1001;
  overflow: hidden;
  font-size: 30px;
  transition: 500ms ease;
`;
