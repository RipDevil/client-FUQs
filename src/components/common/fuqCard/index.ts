export { FuqCard } from './fuq-card';
