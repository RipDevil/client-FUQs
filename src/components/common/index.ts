export { Credits } from './credits';
export { FuqCard } from './fuqCard';
export { Spinner } from './spinner';
