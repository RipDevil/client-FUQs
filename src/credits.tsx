import * as React from 'react';
import styled from 'styled-components';

const CreditsComponent = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1001;
  overflow: hidden;
  font-size: 10px;
  transition: 500ms ease;
  &:hover {
    font-size: 12px;
    bacground-color: lightyellow;
  }
`;

const Credits: React.FC = () => {
  const development = process.env.NODE_ENV === 'development';
  return development ? (
    <CreditsComponent>
      <span>
        <a href='https://github.com/RipDevil' title='RipDevil github page'>
          Github
        </a>
        <a
          href='https://github.com/users/RipDevil/projects/1'
          title='Projects in github'>
          /fuq
        </a>
        <a
          href='https://github.com/RipDevil/client-FUQs'
          title='Client repo in github'>
          /cli
        </a>
      </span>
      <br />
      <span>
        <a href='https://ripdevil.github.io/#/' title='Github pages'>
          WebInstance
        </a>
      </span>
    </CreditsComponent>
  ) : null;
};

export default Credits;
