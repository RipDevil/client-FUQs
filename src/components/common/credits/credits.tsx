import * as React from 'react';

import { CreditsComponent } from './credits.styled';

interface CreditsTypes {
  env: string;
}

export const Credits: React.FC<CreditsTypes> = ({ env }) => {
  return env === 'development' ? (
    <CreditsComponent>
      <span>
        <a href="https://github.com/RipDevil" title="RipDevil github page">
          Github
        </a>
        <a href="https://github.com/users/RipDevil/projects/1" title="Projects in github">
          /fuq
        </a>
        <a href="https://github.com/RipDevil/client-FUQs" title="Client repo in github">
          /cli
        </a>
      </span>
      <br />
      <span>
        <a href="https://ripdevil.github.io/#/" title="Github pages">
          WebInstance
        </a>
      </span>
    </CreditsComponent>
  ) : null;
};
