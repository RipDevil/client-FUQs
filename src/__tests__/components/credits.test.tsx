/* eslint-disable */
import React from 'react';
import { render } from '@testing-library/react';
import { Credits } from 'components/common/credits';

describe('Credits tests', () => {
  it('Should be rendered if in development', () => {
    const { getByText } = render(<Credits env={'development'} />);
    const [gitButton, fuqPrButton, repoButton, webInstanceLink] = [
      getByText(/Github/i),
      getByText(/\/fuq/i),
      getByText(/\/cli/i),
      getByText(/WebInstance/i),
    ];

    expect(gitButton).toBeInTheDocument();
    expect(gitButton).toMatchSnapshot();

    expect(fuqPrButton).toBeInTheDocument();
    expect(fuqPrButton).toMatchSnapshot();

    expect(repoButton).toBeInTheDocument();
    expect(repoButton).toMatchSnapshot();

    expect(webInstanceLink).toBeInTheDocument();
    expect(webInstanceLink).toMatchSnapshot();
  });

  it('Should not be rendered if not in development mode', () => {
    const { container } = render(<Credits env={'gibberish'} />);
    const anyLink = container.querySelector('a');

    expect(anyLink).toBeNull();
  });
});
