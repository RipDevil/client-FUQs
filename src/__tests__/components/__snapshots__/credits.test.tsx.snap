// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credits tests Should be rendered if in development 1`] = `
<a
  href="https://github.com/RipDevil"
  title="RipDevil github page"
>
  Github
</a>
`;

exports[`Credits tests Should be rendered if in development 2`] = `
<a
  href="https://github.com/users/RipDevil/projects/1"
  title="Projects in github"
>
  /fuq
</a>
`;

exports[`Credits tests Should be rendered if in development 3`] = `
<a
  href="https://github.com/RipDevil/client-FUQs"
  title="Client repo in github"
>
  /cli
</a>
`;

exports[`Credits tests Should be rendered if in development 4`] = `
<a
  href="https://ripdevil.github.io/#/"
  title="Github pages"
>
  WebInstance
</a>
`;
